src/sentry/integrations/bitbucket_server/integration.py

```diff
@@ -221,7 +221,10 @@ class BitbucketServerIntegration(IntegrationInstallation, RepositoryMixin):
 
     def get_client(self):
         if self.default_identity is None:
-            self.default_identity = self.get_default_identity()
+            try:
```

Comment:
This will just trade the `Identity.DoesNotExist` exception for an `AttributeError` when you call `client.client.get_commits()`. You'll need to catch the error to make the webhook fail gracefully.